sdk/storage/Azure.Storage.Blobs/src/BlockBlobWriteStream.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Azure.Storage.Shared;
+
+namespace Azure.Storage.Blobs
+{
+    internal class BlockBlobWriteStream : WriteStream
+    {
+        private readonly BlockBlobClient _blockBlobClient;
+        private readonly BlobRequestConditions _conditions;
+        private readonly List<string> _blockIds;
+
+        public BlockBlobWriteStream(
+            BlockBlobClient blockBlobClient,
+            int bufferSize,
```

Comment:
I'd make it long in case we support Jumbo Blocks at some point here.

Btw. if we get there we'd probably want to stream content immediately into put block call rather than buffering.

Reply:
I was concerned with the Put Block call timing out.  What if us user attempts to user attempts to use a Stream returned by Open Write to write to a block blob all day?

Reply:
That'd be the reason to use buffers by default as most fool proof approach. However for users wanting to optimize memory utilization we might need to offer option to live stream data instead.
I believe @jaschrep-msft  is doing something very similar in the parallel uploader (how to not buffer if parallelization=1). This could be an improvement later.

Reply:
Well the user can specify a buffer size, so if they want optimize on memory, they can set the buffer size to like 1 KB.

I think holding open a connection for downloads is fine, if the connection is cut, the user's data in storage isn't effected.  That's not the case for writes, and I'm not sure how we recover from timeouts.

Reply:
Buffer size switched to a long.